{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Resources: https://splinter.readthedocs.io/en/latest/#features \n",
    "#http://web.stanford.edu/~zlotnick/TextAsData/Web_Scraping_with_Beautiful_Soup.html\n",
    "#https://www.crummy.com/software/BeautifulSoup/bs4/doc/\n",
    "\n",
    "#This program conducts a batch BLAST of multiple genes from JGI and stores the top ten results for each BLAST."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Section 1: imports the necessary libraries and creates the arrays needed to store information from JGI and BLAST (RUN THIS)\n",
    "#You have to make sure that you have the splinter and BeautifulSoup libraries (reference resources above for download process)\n",
    "from splinter import Browser\n",
    "from bs4 import BeautifulSoup\n",
    "import pandas as pd\n",
    "from pandas import ExcelWriter\n",
    "from pandas import ExcelFile\n",
    "import numpy as np\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Create scaffold spreadsheet with Filtered Community Composition\n",
    "# Make data frames for every sheet\n",
    "xls = pd.ExcelFile('C:/Users/coverney/Desktop/Research/Meta/Data/Filtered community comparison.xlsx')\n",
    "sample_names = ['FCR15b', 'FCFTC3', 'SCH14c', 'SCB14c', 'FCX16c', \n",
    "                'FCS17a', 'FCW18b', 'SCQ17a', 'SCR11c', 'FCP16c', 'SCD17c']\n",
    "dfs = []\n",
    "for index, name in enumerate(sample_names):\n",
    "    temp_df = pd.read_excel(xls, sample_names[index])\n",
    "    dfs.append(temp_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Create DataFrames\n",
    "headers = dfs[0].loc[0, 'Scaffold ID':]\n",
    "scaffold = pd.DataFrame(columns=headers.index)\n",
    "# references \n",
    "references = pd.DataFrame(columns=['sample', 'scaffold id', 'gene #', 'pfam name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "temp_df = dfs[0]\n",
    "temp_pfams = temp_df['GH1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Go through Lineage Phylum column and if there is a nan\n",
    "# and gene count is >= 5, then add info to scaffold and references spreadsheet\n",
    "for index, df in enumerate(dfs[1:]):\n",
    "#for index, df in enumerate(dfs):\n",
    "    current_pfam = 'GH1'\n",
    "    temp_pfams = df['GH1']\n",
    "    temp_genecounts = df['Gene Count']\n",
    "    temp_phyla = df['Lineage Phylum']\n",
    "    for index2, pfam in enumerate(temp_pfams):\n",
    "        if pfam is not np.nan:\n",
    "            current_pfam = pfam\n",
    "        if temp_phyla[index2] is np.nan:\n",
    "            # Check gene count\n",
    "            temp_genecount = temp_genecounts[index2]\n",
    "            if (temp_genecount is not np.nan) and (temp_genecount >= 5):\n",
    "                temp_row = df.loc[index2, 'Scaffold ID':]\n",
    "                scaffold = scaffold.append(temp_row, ignore_index=True)\n",
    "                temp_row2 = {'sample':sample_names[index], 'scaffold id':temp_row[0], 'gene #':index2, 'pfam name':current_pfam}\n",
    "                references = references.append(temp_row2, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "scaffold.to_excel(\"C:/Users/coverney/Desktop/Research/Meta/Data/scaffolds.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Section 2: opens up a Chrome browser (RUN THIS)\n",
    "executable_path = {'executable_path':'C:\\Program Files\\chromedriver.exe'}\n",
    "browser = Browser('chrome', **executable_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "scaffold_cart_url = \"https://img.jgi.doe.gov/cgi-bin/mer/main.cgi?section=ScaffoldCart&page=index\"\n",
    "browser.visit(scaffold_cart_url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "347"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(scaffold['Scaffold ID'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Go through scaffolds in scaffold cart and store urls to gene counts\n",
    "gene_count_urls = []\n",
    "scaffold_ids = []\n",
    "gene_counts = []\n",
    "for i in range(0,len(scaffold['Scaffold ID'].unique())):\n",
    "# for i in range(0,32):\n",
    "    button = browser.find_by_id('yui-rec' + str(i+100))\n",
    "    soup = BeautifulSoup(button.html, \"html.parser\")\n",
    "    links = soup.find_all('a')\n",
    "    gene_count_urls.append(links[2].get('href'))\n",
    "    gene_counts.append(links[2].get_text())\n",
    "    scaffold_ids.append(links[0].get_text())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "347"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gene_count_urls)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_sequences['scaffold id'] = pd.Series(scaffold_ids)\n",
    "gene_sequences.to_excel(\"C:/Users/coverney/Desktop/Research/Meta/Data/gene_sequences1_3.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Section 3: obtains the top three amino acid sequences from the gene list and stores them in the sequences array\n",
    "#The gene_url is the link to the gene list of a particular scaffold (replace this url with your own), you may be prompted to sign into JGI. \n",
    "#If so, sign in and then re-run this section. \n",
    "#Change the number of genes to blast by changing the \"range(0,3)\" section to \"range(0,n)\", n being the number you want\n",
    "def get_sequences(sub_gene_count_urls, sub_scaffold_ids, sub_gene_counts, gene_sequences):\n",
    "    prefix = \"https://img.jgi.doe.gov/cgi-bin/mer/\"\n",
    "    #gene_sequences = pd.DataFrame(columns=['scaffold id', 'num_seqs', 'seq1', 'seq2', 'seq3', 'seq4', 'seq5'])\n",
    "    #for index, url in enumerate(gene_count_urls[:1]):\n",
    "    for index, url in enumerate(sub_gene_count_urls):\n",
    "        temp_sequences = []\n",
    "        count = 0\n",
    "        ind = 0\n",
    "        gene_count = int(sub_gene_counts[index])\n",
    "        while(count < 5 and ind < gene_count):\n",
    "            browser.visit(prefix+url)\n",
    "            button = browser.find_by_id('yui-rec' + str(ind))\n",
    "            soup = BeautifulSoup(button.html, \"html.parser\")\n",
    "            # Make sure Locus Type is CDS\n",
    "            links = soup.find_all('a')\n",
    "            geneid = links[0].get_text()\n",
    "            if soup.get_text()[len(geneid): len(geneid)+3] == 'CDS':\n",
    "                gene_name = soup.get_text()[len(geneid)+3+len(geneid):]\n",
    "                if 'hypothetical protein' not in gene_name:\n",
    "                    gene_url2 = links[0].get('href')\n",
    "                    browser.visit(prefix + gene_url2)\n",
    "                    soup2 = BeautifulSoup(browser.html, \"html.parser\")\n",
    "                    tab = soup2.table\n",
    "                    tabs = tab.find_all('tr')\n",
    "                    gene_url3 = tabs[15].a.get('href')\n",
    "                    browser.visit(prefix + gene_url3)\n",
    "                    soup3 = BeautifulSoup(browser.html, \"html.parser\")\n",
    "                    a = soup3.pre.get_text()\n",
    "                    b = soup3.pre.font.get_text()\n",
    "                    amino = \"\".join(a.rsplit(b))\n",
    "                    amino = amino.replace('\\n', '')\n",
    "                    temp_sequences.append(amino)\n",
    "                    count += 1\n",
    "                else:\n",
    "                    print('hypo protein')\n",
    "            ind += 1\n",
    "        new_row = {'scaffold id':sub_scaffold_ids[index], 'num_seqs':count}\n",
    "        for i in range(0, count):\n",
    "            temp_name = 'seq' + str(i+1)\n",
    "            new_row.update({temp_name:temp_sequences[i]})\n",
    "        gene_sequences = gene_sequences.append(new_row, ignore_index=True)\n",
    "\n",
    "    return gene_sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_sequences = pd.DataFrame(columns=['scaffold id', 'num_seqs', 'seq1', 'seq2', 'seq3', 'seq4', 'seq5'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n",
      "hypo protein\n"
     ]
    }
   ],
   "source": [
    "# group gene_count_urls into groups of 10 and run get_sequences\n",
    "grouped_gene_count_urls = [gene_count_urls[n:n+10] for n in range(0, len(gene_count_urls), 10)]\n",
    "grouped_gene_counts = [gene_counts[n:n+10] for n in range(0, len(gene_counts), 10)]\n",
    "grouped_scaffold_ids = [scaffold_ids[n:n+10] for n in range(0, len(scaffold_ids), 10)]\n",
    "#for index, group in enumerate(grouped_gene_count_urls[0:1]):\n",
    "for index, group in enumerate(grouped_gene_count_urls):\n",
    "    gene_sequences = get_sequences(group, grouped_scaffold_ids[index], grouped_gene_counts[index], gene_sequences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold id</th>\n",
       "      <th>num_seqs</th>\n",
       "      <th>seq1</th>\n",
       "      <th>seq2</th>\n",
       "      <th>seq3</th>\n",
       "      <th>seq4</th>\n",
       "      <th>seq5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ga0136458_100019</td>\n",
       "      <td>5</td>\n",
       "      <td>MLSKKAKYALQACLSLAGEAPGQPILIASLAERDGIPKKFLEIILL...</td>\n",
       "      <td>VIFDMDGTLLDTESIGIRAWVAAFAAHGVTIDRETAILPIGCDHVR...</td>\n",
       "      <td>MPSLAPIPSLRSRIAAFAFGLFAVDSMAIKVDSVLNQYSLYGTDRT...</td>\n",
       "      <td>MQRYAETHYKFRLPWSPLFRPLPEMIVDAPWCAVPGQDVPLFLAVH...</td>\n",
       "      <td>MSQQPLDLAKELDLALSLVKAAEAPILSRFQKKIEVVKKRDNTPVT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ga0136458_100020</td>\n",
       "      <td>5</td>\n",
       "      <td>MTFLRTLALCLVATSAMARDALRTDSTRTGRASFYHARSGQGTCTL...</td>\n",
       "      <td>MPKRTDIHSILIIGSGPIVIGQACEFDYSGTQACKALREEGYRVIL...</td>\n",
       "      <td>VIGALFSVCVFAAPLWAQEKPLSILTEEWAPYNYTENGVVVGFSVE...</td>\n",
       "      <td>MGCFVASQDDVGQKMTVEIWSDLICPWCSIGLARFDKALAAFPHKE...</td>\n",
       "      <td>MPTYPPSTRPAPLEFSTFLRRERVVIALLAATFVLLAWQHWGMKRS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ga0136458_100030</td>\n",
       "      <td>5</td>\n",
       "      <td>MKTKLLYLCTGNSCRSQMAEAWTRQLKGDLYEVYSAGIEAHGLNPN...</td>\n",
       "      <td>MAKYENNAALEELAEVASALGEPKRLHALGLLSHGELCLCDLTEAL...</td>\n",
       "      <td>MSGSPRAPRASRRRIFAKAAVWAAVAFGFATSASAWSVTGTVKNAS...</td>\n",
       "      <td>MRPVFEYLDYRDILKEAYEERKVSSPLFSYRMLAEFLGLDTSNVFR...</td>\n",
       "      <td>MPDLFTYLEYRDYLKDAYEERRQLHAYFSYRFIGNKVGMDSSYLTR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ga0136458_100035</td>\n",
       "      <td>5</td>\n",
       "      <td>ELKKMLDKVLDGLLTEREAKVLRMYYGINYSKEFTFDEIGRELRLT...</td>\n",
       "      <td>MLILRWLASQALKAREILSILVICIFCLWISNSSEDSQKVWRNAFA...</td>\n",
       "      <td>MAAEKNAPEKFVFYMHRMCKEYTNNKVVLNNISLSFYYGAKIGIIG...</td>\n",
       "      <td>VSFPTSFLNSIGTVPAGLLVLATTCFLGKMAGSVKWAGVKLGGAGV...</td>\n",
       "      <td>MRLLATATALPKSTESAEEIARAIGKSPAWVSDRTGVVRRSRATKE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ga0136458_100037</td>\n",
       "      <td>5</td>\n",
       "      <td>VFTRLAHVCLNVKNLDRSIAYYRKLGFEPRFEFTRKGGRFGAYLEI...</td>\n",
       "      <td>MYRKLSALATAVAWSASIAVAGNPLDIHGRLQAKGNRIVGETSGDT...</td>\n",
       "      <td>MQTIRELLDDPRLFPSLRRIVPSRPRLDVAGLDGAAPAVSIAMRHL...</td>\n",
       "      <td>VIKPELLAPAGGWDCAKAAIENGADAIYFGSEVFNARMRADNFTLA...</td>\n",
       "      <td>LSSSKPEGDTRKKWGNFGEERAASLLRARGLDVIDRNWRHGRGELD...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        scaffold id num_seqs  \\\n",
       "0  Ga0136458_100019        5   \n",
       "1  Ga0136458_100020        5   \n",
       "2  Ga0136458_100030        5   \n",
       "3  Ga0136458_100035        5   \n",
       "4  Ga0136458_100037        5   \n",
       "\n",
       "                                                seq1  \\\n",
       "0  MLSKKAKYALQACLSLAGEAPGQPILIASLAERDGIPKKFLEIILL...   \n",
       "1  MTFLRTLALCLVATSAMARDALRTDSTRTGRASFYHARSGQGTCTL...   \n",
       "2  MKTKLLYLCTGNSCRSQMAEAWTRQLKGDLYEVYSAGIEAHGLNPN...   \n",
       "3  ELKKMLDKVLDGLLTEREAKVLRMYYGINYSKEFTFDEIGRELRLT...   \n",
       "4  VFTRLAHVCLNVKNLDRSIAYYRKLGFEPRFEFTRKGGRFGAYLEI...   \n",
       "\n",
       "                                                seq2  \\\n",
       "0  VIFDMDGTLLDTESIGIRAWVAAFAAHGVTIDRETAILPIGCDHVR...   \n",
       "1  MPKRTDIHSILIIGSGPIVIGQACEFDYSGTQACKALREEGYRVIL...   \n",
       "2  MAKYENNAALEELAEVASALGEPKRLHALGLLSHGELCLCDLTEAL...   \n",
       "3  MLILRWLASQALKAREILSILVICIFCLWISNSSEDSQKVWRNAFA...   \n",
       "4  MYRKLSALATAVAWSASIAVAGNPLDIHGRLQAKGNRIVGETSGDT...   \n",
       "\n",
       "                                                seq3  \\\n",
       "0  MPSLAPIPSLRSRIAAFAFGLFAVDSMAIKVDSVLNQYSLYGTDRT...   \n",
       "1  VIGALFSVCVFAAPLWAQEKPLSILTEEWAPYNYTENGVVVGFSVE...   \n",
       "2  MSGSPRAPRASRRRIFAKAAVWAAVAFGFATSASAWSVTGTVKNAS...   \n",
       "3  MAAEKNAPEKFVFYMHRMCKEYTNNKVVLNNISLSFYYGAKIGIIG...   \n",
       "4  MQTIRELLDDPRLFPSLRRIVPSRPRLDVAGLDGAAPAVSIAMRHL...   \n",
       "\n",
       "                                                seq4  \\\n",
       "0  MQRYAETHYKFRLPWSPLFRPLPEMIVDAPWCAVPGQDVPLFLAVH...   \n",
       "1  MGCFVASQDDVGQKMTVEIWSDLICPWCSIGLARFDKALAAFPHKE...   \n",
       "2  MRPVFEYLDYRDILKEAYEERKVSSPLFSYRMLAEFLGLDTSNVFR...   \n",
       "3  VSFPTSFLNSIGTVPAGLLVLATTCFLGKMAGSVKWAGVKLGGAGV...   \n",
       "4  VIKPELLAPAGGWDCAKAAIENGADAIYFGSEVFNARMRADNFTLA...   \n",
       "\n",
       "                                                seq5  \n",
       "0  MSQQPLDLAKELDLALSLVKAAEAPILSRFQKKIEVVKKRDNTPVT...  \n",
       "1  MPTYPPSTRPAPLEFSTFLRRERVVIALLAATFVLLAWQHWGMKRS...  \n",
       "2  MPDLFTYLEYRDYLKDAYEERRQLHAYFSYRFIGNKVGMDSSYLTR...  \n",
       "3  MRLLATATALPKSTESAEEIARAIGKSPAWVSDRTGVVRRSRATKE...  \n",
       "4  LSSSKPEGDTRKKWGNFGEERAASLLRARGLDVIDRNWRHGRGELD...  "
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_sequences.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "347"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_sequences.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_sequences.to_excel(\"C:/Users/coverney/Desktop/Research/Meta/Data/remaining_gene_sequences1.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold id</th>\n",
       "      <th>num_seqs</th>\n",
       "      <th>seq1</th>\n",
       "      <th>seq2</th>\n",
       "      <th>seq3</th>\n",
       "      <th>seq4</th>\n",
       "      <th>seq5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ga0136458_100019</td>\n",
       "      <td>5</td>\n",
       "      <td>MLSKKAKYALQACLSLAGEAPGQPILIASLAERDGIPKKFLEIILL...</td>\n",
       "      <td>VIFDMDGTLLDTESIGIRAWVAAFAAHGVTIDRETAILPIGCDHVR...</td>\n",
       "      <td>MPSLAPIPSLRSRIAAFAFGLFAVDSMAIKVDSVLNQYSLYGTDRT...</td>\n",
       "      <td>MQRYAETHYKFRLPWSPLFRPLPEMIVDAPWCAVPGQDVPLFLAVH...</td>\n",
       "      <td>MSQQPLDLAKELDLALSLVKAAEAPILSRFQKKIEVVKKRDNTPVT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ga0136458_100020</td>\n",
       "      <td>5</td>\n",
       "      <td>MTFLRTLALCLVATSAMARDALRTDSTRTGRASFYHARSGQGTCTL...</td>\n",
       "      <td>MPKRTDIHSILIIGSGPIVIGQACEFDYSGTQACKALREEGYRVIL...</td>\n",
       "      <td>VIGALFSVCVFAAPLWAQEKPLSILTEEWAPYNYTENGVVVGFSVE...</td>\n",
       "      <td>MGCFVASQDDVGQKMTVEIWSDLICPWCSIGLARFDKALAAFPHKE...</td>\n",
       "      <td>MPTYPPSTRPAPLEFSTFLRRERVVIALLAATFVLLAWQHWGMKRS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ga0136458_100030</td>\n",
       "      <td>5</td>\n",
       "      <td>MKTKLLYLCTGNSCRSQMAEAWTRQLKGDLYEVYSAGIEAHGLNPN...</td>\n",
       "      <td>MAKYENNAALEELAEVASALGEPKRLHALGLLSHGELCLCDLTEAL...</td>\n",
       "      <td>MSGSPRAPRASRRRIFAKAAVWAAVAFGFATSASAWSVTGTVKNAS...</td>\n",
       "      <td>MRPVFEYLDYRDILKEAYEERKVSSPLFSYRMLAEFLGLDTSNVFR...</td>\n",
       "      <td>MPDLFTYLEYRDYLKDAYEERRQLHAYFSYRFIGNKVGMDSSYLTR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ga0136458_100035</td>\n",
       "      <td>5</td>\n",
       "      <td>ELKKMLDKVLDGLLTEREAKVLRMYYGINYSKEFTFDEIGRELRLT...</td>\n",
       "      <td>MLILRWLASQALKAREILSILVICIFCLWISNSSEDSQKVWRNAFA...</td>\n",
       "      <td>MAAEKNAPEKFVFYMHRMCKEYTNNKVVLNNISLSFYYGAKIGIIG...</td>\n",
       "      <td>VSFPTSFLNSIGTVPAGLLVLATTCFLGKMAGSVKWAGVKLGGAGV...</td>\n",
       "      <td>MRLLATATALPKSTESAEEIARAIGKSPAWVSDRTGVVRRSRATKE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ga0136458_100037</td>\n",
       "      <td>5</td>\n",
       "      <td>VFTRLAHVCLNVKNLDRSIAYYRKLGFEPRFEFTRKGGRFGAYLEI...</td>\n",
       "      <td>MYRKLSALATAVAWSASIAVAGNPLDIHGRLQAKGNRIVGETSGDT...</td>\n",
       "      <td>MQTIRELLDDPRLFPSLRRIVPSRPRLDVAGLDGAAPAVSIAMRHL...</td>\n",
       "      <td>VIKPELLAPAGGWDCAKAAIENGADAIYFGSEVFNARMRADNFTLA...</td>\n",
       "      <td>LSSSKPEGDTRKKWGNFGEERAASLLRARGLDVIDRNWRHGRGELD...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        scaffold id  num_seqs  \\\n",
       "0  Ga0136458_100019         5   \n",
       "1  Ga0136458_100020         5   \n",
       "2  Ga0136458_100030         5   \n",
       "3  Ga0136458_100035         5   \n",
       "4  Ga0136458_100037         5   \n",
       "\n",
       "                                                seq1  \\\n",
       "0  MLSKKAKYALQACLSLAGEAPGQPILIASLAERDGIPKKFLEIILL...   \n",
       "1  MTFLRTLALCLVATSAMARDALRTDSTRTGRASFYHARSGQGTCTL...   \n",
       "2  MKTKLLYLCTGNSCRSQMAEAWTRQLKGDLYEVYSAGIEAHGLNPN...   \n",
       "3  ELKKMLDKVLDGLLTEREAKVLRMYYGINYSKEFTFDEIGRELRLT...   \n",
       "4  VFTRLAHVCLNVKNLDRSIAYYRKLGFEPRFEFTRKGGRFGAYLEI...   \n",
       "\n",
       "                                                seq2  \\\n",
       "0  VIFDMDGTLLDTESIGIRAWVAAFAAHGVTIDRETAILPIGCDHVR...   \n",
       "1  MPKRTDIHSILIIGSGPIVIGQACEFDYSGTQACKALREEGYRVIL...   \n",
       "2  MAKYENNAALEELAEVASALGEPKRLHALGLLSHGELCLCDLTEAL...   \n",
       "3  MLILRWLASQALKAREILSILVICIFCLWISNSSEDSQKVWRNAFA...   \n",
       "4  MYRKLSALATAVAWSASIAVAGNPLDIHGRLQAKGNRIVGETSGDT...   \n",
       "\n",
       "                                                seq3  \\\n",
       "0  MPSLAPIPSLRSRIAAFAFGLFAVDSMAIKVDSVLNQYSLYGTDRT...   \n",
       "1  VIGALFSVCVFAAPLWAQEKPLSILTEEWAPYNYTENGVVVGFSVE...   \n",
       "2  MSGSPRAPRASRRRIFAKAAVWAAVAFGFATSASAWSVTGTVKNAS...   \n",
       "3  MAAEKNAPEKFVFYMHRMCKEYTNNKVVLNNISLSFYYGAKIGIIG...   \n",
       "4  MQTIRELLDDPRLFPSLRRIVPSRPRLDVAGLDGAAPAVSIAMRHL...   \n",
       "\n",
       "                                                seq4  \\\n",
       "0  MQRYAETHYKFRLPWSPLFRPLPEMIVDAPWCAVPGQDVPLFLAVH...   \n",
       "1  MGCFVASQDDVGQKMTVEIWSDLICPWCSIGLARFDKALAAFPHKE...   \n",
       "2  MRPVFEYLDYRDILKEAYEERKVSSPLFSYRMLAEFLGLDTSNVFR...   \n",
       "3  VSFPTSFLNSIGTVPAGLLVLATTCFLGKMAGSVKWAGVKLGGAGV...   \n",
       "4  VIKPELLAPAGGWDCAKAAIENGADAIYFGSEVFNARMRADNFTLA...   \n",
       "\n",
       "                                                seq5  \n",
       "0  MSQQPLDLAKELDLALSLVKAAEAPILSRFQKKIEVVKKRDNTPVT...  \n",
       "1  MPTYPPSTRPAPLEFSTFLRRERVVIALLAATFVLLAWQHWGMKRS...  \n",
       "2  MPDLFTYLEYRDYLKDAYEERRQLHAYFSYRFIGNKVGMDSSYLTR...  \n",
       "3  MRLLATATALPKSTESAEEIARAIGKSPAWVSDRTGVVRRSRATKE...  \n",
       "4  LSSSKPEGDTRKKWGNFGEERAASLLRARGLDVIDRNWRHGRGELD...  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in gene sequences\n",
    "gene_sequences = pd.read_excel('C:/Users/coverney/Desktop/Research/Meta/Data/remaining_gene_sequences1.xlsx')\n",
    "gene_sequences.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MIKLACSALLALAGFSAAIPVDTFTVKGPDSAVSFLARPATRDAIKRPALVLAPEWWGVNGYAQRRARELAAGGYVVLAVDFYGNRKVVADRAEAGKLAGAFYGNPAKFRQILTAAVAQLKARSDVDSTRIGALGFCFGGTAVLEGARAGLPFVATASFHGGVKPFASTAKGAVKGRVLVLHGGADPNVTMEAVTESVKDFEAAGADYGVVVYPHAVHAFTNPDAGNDPTKGVAYDAAAEKASFEAFHRLLAETGLKP*'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_sequences_seqs = gene_sequences.loc[:, 'seq1':]\n",
    "gene_sequences_seqs.iloc[12].values[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "executable_path = {'executable_path':'C:\\Program Files\\chromedriver.exe'}\n",
    "browser = Browser('chrome', **executable_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def wait(browser):\n",
    "    found_title = browser.find_by_text('                            \\n                   BLAST Results\\n                ')\n",
    "    if found_title:\n",
    "        return 1\n",
    "    else:\n",
    "        return 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def blast(seq, browser):\n",
    "    blast_url = \"https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins\"\n",
    "    browser.visit(blast_url)\n",
    "    browser.fill('QUERY', seq)\n",
    "    button = browser.find_by_id('b1')\n",
    "    button.click()\n",
    "    move_on = wait(browser)\n",
    "    while not move_on:\n",
    "        move_on = wait(browser)\n",
    "    table = browser.find_by_id('dscTable')\n",
    "    result = []\n",
    "    if len(table) != 0:\n",
    "        for i in range(1,11):  \n",
    "            try:\n",
    "                res = table.find_by_id('deflnDesc_' + str(i)).html\n",
    "                pos = res.find('[')\n",
    "                result.append(res[pos+1:-1])\n",
    "            except:\n",
    "                break\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def find_phylum(val, browser):\n",
    "    tax_website = 'https://www.ncbi.nlm.nih.gov/taxonomy'\n",
    "    browser.visit(tax_website)\n",
    "    browser.fill('term', val)\n",
    "    button = browser.find_by_id('search')\n",
    "    button.click()\n",
    "    wait = browser.find_by_id('messagearea')\n",
    "    while len(wait) is 0:\n",
    "        wait = browser.find_by_id('messagearea')\n",
    "    search_area = browser.find_by_id('maincontent')\n",
    "    soup = BeautifulSoup(search_area.html, \"html.parser\")\n",
    "    links = soup.find_all('a')\n",
    "    url = ''\n",
    "    for link in links:\n",
    "        if link.get_text() == val:\n",
    "            url = 'https://www.ncbi.nlm.nih.gov/'+link.get('href')\n",
    "            break\n",
    "    if url:\n",
    "        browser.visit(url)\n",
    "        soup = BeautifulSoup(browser.html, \"html.parser\")\n",
    "        try:\n",
    "            phylum = soup.find(title='phylum').get_text()\n",
    "            return phylum\n",
    "        except:\n",
    "            return 'none'\n",
    "    else:\n",
    "        return 'none'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def find_phylum_of_gene(phyla):\n",
    "    counter = Counter(phyla).most_common(1)\n",
    "    limit = len(phyla) // 2 + (len(phyla) % 2 > 0)\n",
    "    if limit == 0:\n",
    "        return 'none'\n",
    "    if counter[0][1] >= limit:\n",
    "        phylum = counter[0][0]\n",
    "    else:\n",
    "        phylum = 'none'\n",
    "    return phylum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def find_phylum_of_scaffold(phyla):\n",
    "    counter = Counter(phyla).most_common(1)\n",
    "    limit = len(phyla) // 2 + (len(phyla) % 2 > 0)\n",
    "    if limit == 0:\n",
    "        return 'Unclassified'\n",
    "    if limit < 2:\n",
    "        limit = 2\n",
    "    if counter[0][0] != 'none' and counter[0][1] >= limit:\n",
    "        phylum = counter[0][0]\n",
    "    else:\n",
    "        phylum = 'Unclassified'\n",
    "    return phylum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def find_phyla_of_sample(gene_sequences_seqs, rng, num_seqs):\n",
    "    overall_phyla = []\n",
    "    # Go through every scaffold\n",
    "    for index in rng:\n",
    "        scaffold_phyla = []\n",
    "        # Get the gene sequences of the scaffold\n",
    "        sequences = gene_sequences_seqs.iloc[index].values\n",
    "        # Get the number of sequences\n",
    "        num_seq = num_seqs[index]\n",
    "        # Go through the sequences and BLAST\n",
    "        index2 = 0\n",
    "        three_phyla = 0\n",
    "        while index2 < num_seq and three_phyla < 3:\n",
    "            seq = sequences[index2]\n",
    "            blast_results = blast(seq, browser) # results for one gene\n",
    "            if blast_results:\n",
    "                gene_phyla = []\n",
    "                five_phyla = 0\n",
    "                index3 = 0\n",
    "                while index3 < len(blast_results) and five_phyla < 5:\n",
    "                    elm = blast_results[index3]\n",
    "                    phylum = find_phylum(elm, browser)\n",
    "                    if phylum != 'none':\n",
    "                        gene_phyla.append(phylum)\n",
    "                        five_phyla += 1\n",
    "                    index3 += 1\n",
    "                # Find phylum of gene \n",
    "                gene_phylum = find_phylum_of_gene(gene_phyla)\n",
    "                # print(gene_phylum)\n",
    "                # If phyla is not none then increment three_phyla\n",
    "                if gene_phylum != 'none':\n",
    "                    three_phyla += 1\n",
    "                    scaffold_phyla.append(gene_phylum)\n",
    "            index2 += 1\n",
    "        # Find phylum of scaffold\n",
    "        scaffold_phylum = find_phylum_of_scaffold(scaffold_phyla)\n",
    "        print(scaffold_phylum)\n",
    "        overall_phyla.append(scaffold_phylum)\n",
    "    return overall_phyla"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Firmicutes\n"
     ]
    }
   ],
   "source": [
    "#overall_phyla = find_phyla_of_sample(gene_sequences_seqs, range(0,gene_sequences_seqs.shape[0]))\n",
    "overall_phyla = find_phyla_of_sample(gene_sequences_seqs, range(74,75), gene_sequences['num_seqs'])\n",
    "#print(overall_phyla)\n",
    "#gene_sequences['phylum'] = pd.Series(overall_phyla)\n",
    "#gene_sequences.to_excel(\"C:/Users/coverney/Desktop/Research/Meta/Data/gene_sequences2.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_sequences2 = pd.read_excel('C:/Users/coverney/Desktop/Research/Meta/Data/gene_sequences1_3.xlsx')\n",
    "gene_sequences2['phyla_temp'].head()\n",
    "phyla_temp = gene_sequences2['phyla_temp'].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "phyla_temp = list(filter(lambda a: a != 'none', phyla_temp))\n",
    "subphyla_temp = [phyla_temp[n:n+3] for n in range(0, len(phyla_temp), 3)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Bacteroidetes', 'Bacteroidetes']"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subphyla_temp[-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unclassified\n",
      "Proteobacteria\n",
      "Unclassified\n",
      "Proteobacteria\n",
      "Bacteroidetes\n",
      "Unclassified\n",
      "Fibrobacteres\n",
      "Proteobacteria\n",
      "Proteobacteria\n",
      "Unclassified\n",
      "Fibrobacteres\n",
      "Proteobacteria\n",
      "Fibrobacteres\n",
      "Proteobacteria\n",
      "Proteobacteria\n",
      "Proteobacteria\n",
      "Proteobacteria\n",
      "Unclassified\n",
      "Unclassified\n",
      "Unclassified\n",
      "Unclassified\n",
      "Unclassified\n",
      "Unclassified\n",
      "Unclassified\n",
      "Unclassified\n",
      "Bacteroidetes\n",
      "Proteobacteria\n",
      "Firmicutes\n",
      "Fibrobacteres\n"
     ]
    }
   ],
   "source": [
    "for group in subphyla_temp:\n",
    "    if len(group) >= 3:\n",
    "        scaffold_phylum = find_phylum_of_scaffold(group)\n",
    "        print(scaffold_phylum)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Proteobacteria']\n"
     ]
    }
   ],
   "source": [
    "print(overall_phyla)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Reference NCBI Taxonomy website\n",
    "tax_website = 'https://www.ncbi.nlm.nih.gov/taxonomy'\n",
    "all_phylas = []\n",
    "overall_phyla = []\n",
    "# go through scaffolds\n",
    "#for results in gene_sequences['results']:\n",
    "for results in all_results:\n",
    "    phylas = []\n",
    "    # go through genes\n",
    "    for res in results:\n",
    "        # go through results from one gene\n",
    "        phyla = []\n",
    "        for val in res:\n",
    "            browser.visit(tax_website)\n",
    "            browser.fill('term', val)\n",
    "            button = browser.find_by_id('search')\n",
    "            button.click()\n",
    "            wait = browser.find_by_id('messagearea')\n",
    "            while len(wait) is 0:\n",
    "                wait = browser.find_by_id('messagearea')\n",
    "            search_area = browser.find_by_id('maincontent')\n",
    "            soup = BeautifulSoup(search_area.html, \"html.parser\")\n",
    "            links = soup.find_all('a')\n",
    "            for link in links:\n",
    "                if link.get_text() == val:\n",
    "                    url = 'https://www.ncbi.nlm.nih.gov/'+link.get('href')\n",
    "                    break\n",
    "            browser.visit(url)\n",
    "            soup = BeautifulSoup(browser.html, \"html.parser\")\n",
    "            try:\n",
    "                phylum = soup.find(title='phylum').get_text()\n",
    "                phyla.append(phylum)\n",
    "            except:\n",
    "                continue\n",
    "        gene_phylum_counter = Counter(phyla).most_common(1)\n",
    "        if gene_phylum_counter[0][1] >= 3:\n",
    "            gene_phylum = gene_phylum_counter[0][0]\n",
    "        else:\n",
    "            gene_phylum = 'none'\n",
    "        phylas.append(gene_phylum)\n",
    "    top_phylum = Counter(phylas).most_common(1)\n",
    "    if top_phylum[0][0] != 'none' and top_phylum[0][1] >= 3:\n",
    "        overall_phylum = top_phylum[0][0]\n",
    "    else:\n",
    "        overall_phylum = 'Unclassified'\n",
    "    overall_phyla.append(overall_phylum)\n",
    "    all_phylas.append(phylas)\n",
    "print(overall_phyla)\n",
    "# gene_sequences['phyla'] = pd.Series(overall_phyla)\n",
    "# gene_sequences['phylas'] = pd.Series(all_phylas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[['Candidatus Edwardsbacteria bacterium RifOxyC12_full_54_24', 'Desulfobacteraceae bacterium', 'Deltaproteobacteria bacterium HGW-Deltaproteobacteria-15', 'candidate division Zixibacteria bacterium CG_4_9_14_3_um_filter_46_8', 'Ignavibacteria bacterium RBG_13_36_8'], ['Fibrobacter sp. UWEL', 'Fibrobacter sp. UWH8', 'Fibrobacter sp. UWH8', 'Fibrobacter sp. UWOV1', 'Fibrobacter sp. UWCM'], ['Spirochaetes bacterium', 'Spirochaetae bacterium HGW-Spirochaetae-1', 'Turneriella parva', 'Acidobacteria bacterium', 'Thermococcus profundus'], ['Fibrobacter sp. UWB1', 'Fibrobacter sp. UWOV1', 'Fibrobacter', 'Fibrobacter sp. UWB1', 'Fibrobacter sp. UWB8'], ['Fibrobacter sp. UWR2', 'Geminisphaera colitermitum', 'Fibrobacter', 'Fibrobacter sp. UWP2', 'Fibrobacter sp. UWEL']]]\n"
     ]
    }
   ],
   "source": [
    "#Section 4: takes the sequences from section 3 and BLASTs them.\n",
    "blast_url = \"https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins\"\n",
    "all_results = []\n",
    "gene_sequences_seqs = gene_sequences.loc[:, 'seq1':]\n",
    "for index in range(gene_sequences_seqs.shape[0]):\n",
    "# for index in range(0,1):\n",
    "    results = []\n",
    "    row = gene_sequences_seqs.iloc[index].values\n",
    "    for seq in row:\n",
    "        browser.visit(blast_url)\n",
    "        browser.fill('QUERY', seq)\n",
    "        button = browser.find_by_id('b1')\n",
    "        button.click()\n",
    "        wait = browser.find_by_id('summarylists')\n",
    "        while not wait:\n",
    "            wait = browser.find_by_id('summarylists')\n",
    "        table = browser.find_by_id('dscTable')\n",
    "        if len(table) != 0:\n",
    "            result = []\n",
    "            for i in range(1,6):  \n",
    "                try:\n",
    "                    res = table.find_by_id('deflnDesc_' + str(i)).html\n",
    "                    pos = res.find('[')\n",
    "                    result.append(res[pos+1:-1])\n",
    "                except:\n",
    "                    break\n",
    "            results.append(result)\n",
    "    all_results.append(results)\n",
    "#gene_sequences['results'] = pd.Series(all_results)\n",
    "print(all_results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_sequences.to_excel(\"C:/Users/coverney/Desktop/Research/Meta/Data/gene_sequences2.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Fibrobacteres']\n"
     ]
    }
   ],
   "source": [
    "from collections import Counter\n",
    "# Reference NCBI Taxonomy website\n",
    "tax_website = 'https://www.ncbi.nlm.nih.gov/taxonomy'\n",
    "all_phylas = []\n",
    "overall_phyla = []\n",
    "# go through scaffolds\n",
    "#for results in gene_sequences['results']:\n",
    "for results in all_results:\n",
    "    phylas = []\n",
    "    # go through genes\n",
    "    for res in results:\n",
    "        # go through results from one gene\n",
    "        phyla = []\n",
    "        for val in res:\n",
    "            browser.visit(tax_website)\n",
    "            browser.fill('term', val)\n",
    "            button = browser.find_by_id('search')\n",
    "            button.click()\n",
    "            wait = browser.find_by_id('messagearea')\n",
    "            while len(wait) is 0:\n",
    "                wait = browser.find_by_id('messagearea')\n",
    "            search_area = browser.find_by_id('maincontent')\n",
    "            soup = BeautifulSoup(search_area.html, \"html.parser\")\n",
    "            links = soup.find_all('a')\n",
    "            for link in links:\n",
    "                if link.get_text() == val:\n",
    "                    url = 'https://www.ncbi.nlm.nih.gov/'+link.get('href')\n",
    "                    break\n",
    "            browser.visit(url)\n",
    "            soup = BeautifulSoup(browser.html, \"html.parser\")\n",
    "            try:\n",
    "                phylum = soup.find(title='phylum').get_text()\n",
    "                phyla.append(phylum)\n",
    "            except:\n",
    "                continue\n",
    "        gene_phylum_counter = Counter(phyla).most_common(1)\n",
    "        if gene_phylum_counter[0][1] >= 3:\n",
    "            gene_phylum = gene_phylum_counter[0][0]\n",
    "        else:\n",
    "            gene_phylum = 'none'\n",
    "        phylas.append(gene_phylum)\n",
    "    top_phylum = Counter(phylas).most_common(1)\n",
    "    if top_phylum[0][0] != 'none' and top_phylum[0][1] >= 3:\n",
    "        overall_phylum = top_phylum[0][0]\n",
    "    else:\n",
    "        overall_phylum = 'Unclassified'\n",
    "    overall_phyla.append(overall_phylum)\n",
    "    all_phylas.append(phylas)\n",
    "print(overall_phyla)\n",
    "# gene_sequences['phyla'] = pd.Series(overall_phyla)\n",
    "# gene_sequences['phylas'] = pd.Series(all_phylas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[['none', 'Fibrobacteres', 'Spirochaetes', 'Fibrobacteres', 'Fibrobacteres']]\n"
     ]
    }
   ],
   "source": [
    "print(all_phylas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold id</th>\n",
       "      <th>seqs</th>\n",
       "      <th>phylas</th>\n",
       "      <th>results</th>\n",
       "      <th>phyla</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ga0136444_100052</td>\n",
       "      <td>[MATDELLRPTVRLDVFDGPLDLLLYLVHSHELDPRTIPVSQIADQ...</td>\n",
       "      <td>[Fibrobacteres, none, none]</td>\n",
       "      <td>[[Fibrobacter sp. UWS2, Fibrobacter sp. UWS3, ...</td>\n",
       "      <td>Unclassified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ga0136444_100059</td>\n",
       "      <td>[MRIRLVCFGKLVPKPLGPAVEEYAKRLEKLCRLEIHELPEAQGPD...</td>\n",
       "      <td>[Firmicutes, none, none]</td>\n",
       "      <td>[[Clostridium, Clostridium botulinum, Clostrid...</td>\n",
       "      <td>Unclassified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ga0136444_100076</td>\n",
       "      <td>[MKAVVVLALASALVGLGGCATIKVHDAVQEIGLLEPGHLEWWDDS...</td>\n",
       "      <td>[none, none]</td>\n",
       "      <td>[[Pseudomonas, Candidatus Marinimicrobia bacte...</td>\n",
       "      <td>Unclassified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ga0136444_100106</td>\n",
       "      <td>[MRTGLLLAVVALVVLAGFAGARALRSQGGGAVPKEATVRDFDEHG...</td>\n",
       "      <td>[Verrucomicrobia, Bacteroidetes, none]</td>\n",
       "      <td>[[Verrucomicrobia bacterium LW23, Methylacidip...</td>\n",
       "      <td>Unclassified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ga0136444_100122</td>\n",
       "      <td>[PPGAIVPQNGRNPFSSRRWLDAQDRLLPLDSLIPWPGGASIVLRK...</td>\n",
       "      <td>[none, none]</td>\n",
       "      <td>[[Lentzea albida, Lentzea waywayandensis, Delt...</td>\n",
       "      <td>Unclassified</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        scaffold id                                               seqs  \\\n",
       "0  Ga0136444_100052  [MATDELLRPTVRLDVFDGPLDLLLYLVHSHELDPRTIPVSQIADQ...   \n",
       "1  Ga0136444_100059  [MRIRLVCFGKLVPKPLGPAVEEYAKRLEKLCRLEIHELPEAQGPD...   \n",
       "2  Ga0136444_100076  [MKAVVVLALASALVGLGGCATIKVHDAVQEIGLLEPGHLEWWDDS...   \n",
       "3  Ga0136444_100106  [MRTGLLLAVVALVVLAGFAGARALRSQGGGAVPKEATVRDFDEHG...   \n",
       "4  Ga0136444_100122  [PPGAIVPQNGRNPFSSRRWLDAQDRLLPLDSLIPWPGGASIVLRK...   \n",
       "\n",
       "                                   phylas  \\\n",
       "0             [Fibrobacteres, none, none]   \n",
       "1                [Firmicutes, none, none]   \n",
       "2                            [none, none]   \n",
       "3  [Verrucomicrobia, Bacteroidetes, none]   \n",
       "4                            [none, none]   \n",
       "\n",
       "                                             results         phyla  \n",
       "0  [[Fibrobacter sp. UWS2, Fibrobacter sp. UWS3, ...  Unclassified  \n",
       "1  [[Clostridium, Clostridium botulinum, Clostrid...  Unclassified  \n",
       "2  [[Pseudomonas, Candidatus Marinimicrobia bacte...  Unclassified  \n",
       "3  [[Verrucomicrobia bacterium LW23, Methylacidip...  Unclassified  \n",
       "4  [[Lentzea albida, Lentzea waywayandensis, Delt...  Unclassified  "
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_sequences.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_sequences.to_excel(\"C:/Users/coverney/Desktop/Research/Meta/Data/gene_sequences3.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['hypothetical protein [Fibrobacter sp. UWS2]',\n",
       " 'hypothetical protein [Fibrobacter sp. UWS3]',\n",
       " 'chromosome segregation protein ScpA [Fibrobacter sp. UWB12]',\n",
       " 'condensin subunit ScpA [Fibrobacter sp. UWT3]',\n",
       " 'hypothetical protein [Fibrobacter sp. UWOV1]']"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get the actual data\n",
    "result = []\n",
    "table = browser.find_by_id('dscTable')\n",
    "for i in range(1,6):  \n",
    "    res = table.find_by_id('deflnDesc_' + str(i)).html\n",
    "    result.append(res)\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_sequences.to_excel(\"C:/Users/coverney/Desktop/Research/Meta/Data/gene_sequences2.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n",
    "gene_sequences['phylas'] = pd.Series(phylas)\n",
    "for phyla in gene_sequences['phylas']:\n",
    "    counter = Counter(phyla)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# get the taxonomy report\n",
    "        soup = BeautifulSoup(browser.html, \"html.parser\")\n",
    "        links = soup.find_all('a')\n",
    "        for link in links:\n",
    "            if link.get_text() == '[Taxonomy reports]':\n",
    "                tax_url = link.get('href')\n",
    "\n",
    "        browser.visit('https://blast.ncbi.nlm.nih.gov/'+tax_url)\n",
    "        # get phyla\n",
    "        table = browser.find_by_id('lngTable')\n",
    "        soup = BeautifulSoup(table.html, \"html.parser\")\n",
    "        links = soup.find_all('a')\n",
    "        phylum = links[0].get_text()\n",
    "        if phylum != 'cellular organisms' and phylum != 'Bacteria':\n",
    "            phyla.append(phylum)\n",
    "        else:\n",
    "            phyla.append('none')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
